deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/EarlyTerminationMultiDataSetIterator.java

```diff
@@ -0,0 +1,87 @@
+package org.deeplearning4j.datasets.iterator;
+
+import org.nd4j.linalg.dataset.api.MultiDataSet;
+import org.nd4j.linalg.dataset.api.MultiDataSetPreProcessor;
+import org.nd4j.linalg.dataset.api.iterator.MultiDataSetIterator;
+
+/**
+ * Builds an iterator that terminates once the number of minibatches returned with .next() is equal to a specified number
+ * Note that a call to .next(num) is counted as a call to return a minibatch regardless of the value of num
+ * This essentially restricts the data to this specified number of minibatches.
+ */
+public class EarlyTerminationMultiDataSetIterator implements MultiDataSetIterator {
+
+    private MultiDataSetIterator underlyingIterator;
+    private int terminationPoint;
+    private int minibatchCount = 0;
+
+    /**
+     * Constructor takes the iterator to wrap and the number of minibatches after which the call to hasNext()
+     * will return false
+     * @param underlyingIterator, iterator to wrap
+     * @param terminationPoint, minibatches after which hasNext() will return false
+     */
+    public EarlyTerminationMultiDataSetIterator(MultiDataSetIterator underlyingIterator, int terminationPoint) {
+        if (terminationPoint <= 0 ) throw new IllegalArgumentException("Termination point (the number of calls to .next() or .next(num)) has to be > 0");
+        this.underlyingIterator = underlyingIterator;
+        this.terminationPoint = terminationPoint;
+    }
+
+    @Override
+    public MultiDataSet next(int num) {
```

Comment:
The termination point is not the number of elements. It is the number of minibatches. I have made that clear in the javadoc. 
https://github.com/deeplearning4j/deeplearning4j/pull/3502/files#diff-b986a13d8f51bf0dae2466fc7faa4d52R11